x-pack/plugins/alerting/server/create_fire_handler.test.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createFireHandler } from './create_fire_handler';
+
+const createFireHandlerParams = {
+  fireAction: jest.fn(),
+  alertSavedObject: {
+    id: '1',
+    type: 'alert',
+    attributes: {
+      alertTypeId: '123',
+      interval: 10000,
+      alertTypeParams: {
+        bar: true,
+      },
+      actions: [
+        {
+          group: 'default',
+          actionRef: 'action_0',
+          params: {
+            foo: true,
+            contextVal: 'My {{context.value}} goes here',
+            stateVal: 'My {{state.value}} goes here',
+          },
+        },
+      ],
+    },
+    references: [
+      {
+        name: 'action_0',
+        type: 'action',
+        id: '1',
+      },
+    ],
+  },
+};
+
+beforeEach(() => jest.resetAllMocks());
+
+test('calls fireAction per selected action', async () => {
+  const fireHandler = createFireHandler(createFireHandlerParams);
+  await fireHandler('default', {}, {});
+  expect(createFireHandlerParams.fireAction).toMatchInlineSnapshot(`
+[MockFunction] {
```

Comment:
Kind of the same issues as above: these are hard to read with long inline snapshots and I question whether they're really testing all that much.